### ISSUE TITLE
[V3] Unmute has odd behaviour on admins

### ISSUE THREAD

**Issue by coltoutram** [Cog-Creators/Red-DiscordBot#2076]
# Command bugs

#### Command name

Unmute

#### What cog is this command from?

Mod

#### What were you expecting to happen?

Unmute someone from the server.

#### What actually happened?

It sent the "user has been unmuted from the server" but did not remove the permissions from the channels.

#### How can we reproduce this issue?

Mute someone using [p]mute server user and then [p]unmute server user and it will not remove the permissions from the channels.

**Comment by coltoutram** [comment]
What kind of info is needed?

**Comment by Ray-MRQ** [comment]
traceback error... if there is one

edit: "but did not remove the permissions from the channels." that doesn't mean they weren't unmuted it just means the user profile on the channel wasn't removed if you check it the write perm was re-enabled (if i understood that correctly)

**Comment by Zephyrkul** [comment]
I've been playing around with this a little. This is in a test server with no special channel overwrites beyond an admin role that has access to a private category, text channel, and voice channel. This isn't full QA by any means but it's what I found while messing around with it.

Mute:
For every channel the bot and member have access to...
Categories and VCs are denied the Speak permission.
Text channels are denied Send Messages and Add Reactions.
(As a small note this causes categories and their text channels to no longer be in sync.)
No traceback in console.

Unmute:
I got a couple different results depending on the member's permissions.
If the member has the admin override, nothing happens. Overwrites set via Mute are as they were.
If the member has no admin override, text channel overwrites are deleted, while category and VC overwrites for that member are set to all None (including for categories / VCs that were not touched by mute beforehand).
The bot having a higher role than the member or not has no effect on the above behavior.
No traceback in console in either case.

**Comment by coltoutram** [comment]
The bot has the highest role and none of the users I am trying to mute have a role above the BOT, I'm confused why mute works but unmute does not, any info?

**Comment by Zephyrkul** [comment]
Literally that big ol' comment directly above yours was a huge infodump when I tested this issue out. Give it a read if you want info.

**Comment by coltoutram** [comment]
Yes I read that, from what I understand it does not work the way it is intended to?

**Comment by Zephyrkul** [comment]
As it stands, based on my testing, `[p]unmute` does *function*, though it could use improvements. In my testing it only ever failed to unmute members who had the Administrator override in their permissions, which allows them to bypass channel permissions. So muting/unmuting doesn't affect them either way. Since currently the bot only unmutes in channels the member can't send messages to, it doesn't change permissions for Administrator members.

In other words, since the muted member was an Administrator, they were never muted in the first place, so there's nothing to unmute, so the bot didn't touch channel overwrites.

**Comment by coltoutram** [comment]
I see now, I'm confused why it mutes people with administrator but then does not unmute them though

**Comment by Tobotimus** [comment]
Resolved by #2230.
